app/Http/Controllers/SportsController.php

```diff
@@ -16,17 +17,32 @@ public function create()
         return view('sports.create', compact('sports', 'countries'));
     }
 
-    public function store(Request $request)
-    {
-        // Add your code here
+    public function store(UpdateRanksRequest $request)
+    {   
+        $requests = collect($request->except('_token'));
+
+        $requests->each(function($code, $key) {
+            list($rank, $sportId) = explode('--', $key);
+            $country = Country::where('short_code', $code)->first();
```

Comment:
@jstarnate I think you could get all countries upfront and then get each country from that collection, instead of calling DB query for each of `$requests`

Reply:
@PovilasKorop I should have used `filter()` and populated the winners inside `attach()` :man_facepalming: